### ISSUE TITLE
Why can't `textarea` elements have role `combobox`?

### ISSUE THREAD

**Issue by nolandanley** [dequelabs/axe-core#3548]
Just wanted to ask why `textarea` elements aren't allowed to have the role `combobox`. It seems like just [certain types of `input` elements are allowed](https://github.com/dequelabs/axe-core/issues/1777), so the following code snippet (obviously not the full widget) throws an [`aria-allowed-role`](https://dequeuniversity.com/rules/axe/4.4/aria-allowed-role) violation. Does using text field elements other than `input` cause known accessibility issues? Thanks for any insight!

```<textarea role="combobox"></textarea>```

Product: axe-core v4.4.1

**Comment by WilcoFiers** [comment]
This rule is based on the W3C's [ARIA in HTML](https://www.w3.org/TR/html-aria/#el-textarea) specification, which outlines which HTML elements can have which roles / attributes. For `textarea`, no role is allowed. Axe-core allows the implicit role (`textbox`), there are some use cases for that.

I'm not sure what a multi-line combobox would even look like. Do you have one somewhere? I'd be very interested to see it. Sometimes new ideas on how to use ARIA does come up.

**Comment by nolandanley** [comment]
Sorry for the major delay here! I've seen it used in some instances where there could be a long list of names being types and some autocomplete suggestions could come up for the name you're currently adding. I created a super basic example of this that I'll post below. The autocomplete suggestions will come up if you start with A, B, or C. I didn't give it full accessibility support, not even handling for mouse controls, but this should show the basic keyboard functionality I've seen some developers go for.

```html
<!-- 
  Copyright 2023 Google LLC.
  SPDX-License-Identifier: Apache-2.0
-->

<style>
  #suggestions {
    border: solid;
    list-style: none;
    margin: 0;
    padding: 0;
    width: 100px;
  }

  [role=combobox][aria-expanded=false] + #suggestions {
    display: none;
  }

  [role=option] {
    padding: .5rem;
  }

  [role=option].hidden {
    display: none;
  }

  [role=option]:focus {
    background-color: yellow;
  }
</style>

<script>
  let textarea;
  let suggestionList;
  let suggestionItems;

  function onTextareaKeyup() {
    textarea.focus();
    textarea.setAttribute('aria-expanded', false);
    let currentName = 
textarea.value.split(',').pop().trim().toLowerCase();
    if (currentName.length === 0) return;
    for (const suggestion of suggestionItems) {
      const suggestionValue = suggestion.innerText.toLowerCase();
      if (suggestionValue.startsWith(currentName) &&
          suggestionValue !== currentName) {
        suggestion.classList.remove('hidden');
        textarea.setAttribute('aria-expanded', true);
      } else {
        suggestion.classList.add('hidden');
      }
    }
  }

  function onOptionKeyup(e) {
    if (e.key !== 'Enter') return;
    let value = textarea.value;
    const indexToReplace = value.lastIndexOf(',') + 1;
    value = value.substring(0, indexToReplace);
    if (value.length > 0) value += ' ';
    value += e.target.innerText;
    textarea.value = value;
    onTextareaKeyup();
  }

  function onLoad() {
    textarea = document.querySelector('[role=combobox]');
    textarea.addEventListener('keyup', onTextareaKeyup);
    suggestionList = document.getElementById('suggestions');
    suggestionItems =
      Array.from(suggestionList.querySelectorAll('[role=option]'));
    suggestionItems.forEach(suggestion =>
      suggestion.addEventListener('keyup', onOptionKeyup));
  }
  
  window.addEventListener('load', onLoad);
</script>

<p id="label">Enter comma-separated names</p>
<p id="desc">
  When typing a name, completion suggestions may appear.<br/>
  Tab into them and press enter to select. Mouse controls not supported.
</p>
<textarea role="combobox" aria-labelledby="label"
  aria-describedby="desc" aria-autocomplete="list"
  aria-haspopup="listbox" aria-controls="suggestions"
  aria-expanded="false"></textarea>
<ul id="suggestions" role="listbox">
  <li role="option" tabindex="0">Al Chen</li>
  <li role="option" tabindex="0">Annie Smith</li>
  <li role="option" tabindex="0">Bert Jones</li>
  <li role="option" tabindex="0">Chris Doe</li>
</ul>
```

**Comment by brgibson** [comment]
@WilcoFiers,

One example of a multi-line combo box can be found in the main search bar on Google (although you'll notice the role="combobox" is added to a wrapper div).

<img width="1516" alt="image" src="https://github.com/user-attachments/assets/38bc5868-2291-40ea-9692-7832414e81e1">

We implemented something similar for our main search bar, except we applied the role="combobox" on our textarea (since it is normally added to the input element).  However, once we did that, we noticed this test failing.

**Comment by brgibson** [comment]
@bdugan14.  Thanks for creating https://github.com/w3c/html-aria/issues/543 🙇 🙇 🙇
